src/main/java/org/broadinstitute/hellbender/tools/spark/utils/LongHopscotchSet.java

```diff
@@ -0,0 +1,681 @@
+    private int size;
+    private long[] buckets;
+    // format of the status bytes:
+    // high bit set indicates that the bucket contains a "chain head" (i.e., an entry that naturally belongs in the
+    // corresponding bucket).  high bit not set indicates a "squatter" (i.e., an entry that got placed here through the
+    // collision resolution methodology).  we use Byte.MIN_VALUE (i.e., 0x80) to pick off this bit.
+    // low 7 bits give the (unsigned) offset from the current entry to the next entry in the collision resolution chain.
+    // if the low 7 bits are 0, then we'd be pointing at ourselves, which is nonsense, so that particular value marks
+    // "end of chain" instead.  we use Byte.MAX_VALUE (i.e., 0x7f) to pick off these bits.
+    private byte[] status;
+
+    /**
+     * make a small LongHopscotchSet
+     */
+    public LongHopscotchSet() {
+        this(12000);
+    }
+
+    /**
+     * make a LongHopscotchSet for a specified minimum capacity
+     */
+    public LongHopscotchSet(final int capacity) {
+        this.capacity = getLegalSizeAbove(capacity);
+        this.size = 0;
+        // Not unsafe, because the remainder of the API allows only elements known to be long's to be assigned to buckets.
+        this.buckets = new long[this.capacity];
+        this.status = new byte[this.capacity];
+    }
+
+    /**
+     * make a LongHopscotchSet containing an initial array of values
+     */
+    public LongHopscotchSet(final long[] values) {
+        this.capacity = getLegalSizeAbove(values.length);
+        this.size = 0;
+        // Not unsafe, because the remainder of the API allows only elements known to be long's to be assigned to buckets.
+        this.buckets = new long[this.capacity];
+        this.status = new byte[this.capacity];
+        for (final long val : values) {
+            add(val);
+        }
+    }
+
+    protected LongHopscotchSet(final Kryo kryo, final Input stream) {
+        final boolean oldReferences = kryo.getReferences();
+        kryo.setReferences(false);
+
+        capacity = stream.readInt();
+        size = 0;
+        buckets = new long[capacity];
+        status = new byte[capacity];
+        int nElements = stream.readInt();
+        while (nElements-- > 0) {
+            add(stream.readLong());
+        }
+
+        kryo.setReferences(oldReferences);
+    }
+
+    /**
+     * Computes next largest legal table size for a given number of elements, accounting for load factor
+     */
+    protected static int getLegalSizeAbove(final long minElements) {
+        final long augmentedSize = (long) (minElements / LOAD_FACTOR);
+        for (final int legalSize : legalSizes) {
+            if (legalSize >= augmentedSize) return legalSize;
+        }
+        throw new IllegalArgumentException("No legal sizes large enough for size " + minElements);
+    }
+
+    /**
+     * Computes next smallest legal table size for a given number of elements, accounting for load factor
```

Comment:
Done, although I think it is getting the smallest table size greater than a specified capacity?